atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -130,6 +151,44 @@ public void close() {
         return getRandomGoodHost().runWithPooledResource(CassandraUtils.getDescribeRing(config));
     }
 
+    private String getSnitch() throws Exception {
+        return getRandomGoodHost().runWithPooledResource(
+                (FunctionCheckedException<CassandraClient, String, Exception>) CassandraClient::describe_snitch);
+    }
+
+    private Set<InetSocketAddress> refreshLocalHosts(List<TokenRange> tokenRanges) {
+        Optional<HostLocation> myLocation = myLocationSupplier.get();
+
+        if (!myLocation.isPresent()) {
+            return ImmutableSet.of();
+        }
+
+        Set<InetSocketAddress> newLocalHosts = Sets.newHashSet();
+        for (TokenRange tokenRange : tokenRanges) {
```

Comment:
you should prefer declarative programming to imperative programming in general - easier to read. So you could write this as:

```java
return tokenRanges.stream()
    .map(TokenRange::getEndpoint_details)
    .flatMap(Collection::stream)
    .filter(details -> isHostLocal(details, myLocation.get())
    .map(EndpointDetails::getHost)
    .map(this::getAddressForHostThrowUnchecked)
    .collect(toList());
```